third_party/docfx-doclet-143274/src/main/java/com/microsoft/util/ElementUtil.java

```diff
@@ -15,26 +17,46 @@
 
 public class ElementUtil {
 
+    private static final int INITIAL_CAPACITY = 1000000;
     private final Set<Pattern> excludePackages = new HashSet<>();
     private final Set<Pattern> excludeClasses = new HashSet<>();
 
+    private final Map<Element, List<? extends Element>> elementMap;
+    private final Map<Element, List<TypeElement>> elementSortedMap;

```

Comment:
Create map to cache the element and the results from getEnclosedElements(). From what I see, most of the slowdown is from element.getEnclosedElement().